receiver/redisreceiver/config.md

```diff
@@ -0,0 +1,267 @@
+# Redis Receiver Reference
+
+
+
+### redisreceiver-Config
+
+| Name | Field Info | Default | Docs |
+| ---- | --------- | ------- | ---- |
+| collection_interval |[time-Duration](#time-Duration)| 10s |  |
+| endpoint |string| <no value> |  |
+| transport |string| tcp |  |
+| password |string| <no value> | Optional password. Must match the password specified in the requirepass server configuration option.  |
+| tls |[tls-TLSClientSetting](#tls-TLSClientSetting)| <no value> |  |
+| metrics |[metrics-MetricsSettings](#metrics-MetricsSettings)| <no value> | MetricsSettings provides settings for redisreceiver metrics.  |
+
+### tls-TLSClientSetting
```

Comment:
This would be shared by many components; it would be great to have this at a single place (on the configtls folder over at `opentelemetry-collector`?) to signal that the same section is used by the components

Reply:
Interesting -- would require linking across repos but I suppose that could be taken care of via a config. Wondering if we'd end up writing a markdown file for every type that's used in a config.

Reply:
Even if we keep it in one file for now, do we really need the field name prefix? Can it be just `TLSClientSetting` for better readability?

Reply:
e.g. it would be awesome if we can have one `MetricsSettings` section instead of all the `...-MetricsSettings` sections

Reply:
Let's keep this for later improvements.